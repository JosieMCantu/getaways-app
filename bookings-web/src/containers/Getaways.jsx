import React from 'react';
import PlaceList from '../components/places/PlaceList';
import {useGetaways} from '../hooks/useGetaways';

const Getaways = () => {
  const {places, loading} = useGetaways();

  if(loading) { 
    return <h1>Loading...</h1>
}
  return <PlaceList places={places} />;
};

export default Getaways;
