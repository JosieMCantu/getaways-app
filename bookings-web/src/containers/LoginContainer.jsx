import React, {useState} from 'react';
import Login from '../components/auth/Login';
import loginUser from '../services/authUtils';

const LoginContainer = () => {

    const [userName, setuserName] = useState('');
    const [password, setPassword] = useState('');
    const [email, setEmail] = useState('');

    const onChange = async ({target}) => {
        if(target.name === 'name') setuserName(target.value);
        if(target.name === 'password') setPassword(target.value);
        if(target.name === 'email') setEmail(target.value);
    }

    const handleSubmit = async (event) => {
        event.preventDefault();
        await loginUser(userName, password, email);
        window.location.href = '/';
    }

    return (
        <>
        <Login username={userName} password={password} email={email} onChange={onChange} onSubmit={handleSubmit} />
        </>
    )
}

export default LoginContainer;
