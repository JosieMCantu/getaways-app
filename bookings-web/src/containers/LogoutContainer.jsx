import React, {useState} from 'react';

const LogoutContainer = () => {
    return (
        <>
            
        </>
    )
}

export default LogoutContainer;