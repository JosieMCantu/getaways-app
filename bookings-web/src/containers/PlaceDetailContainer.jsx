import React, {useState, useEffect} from 'react';
import {useParams} from 'react-router-dom';
import {getPlaceById} from '../services/placesApi';
import PlaceDetail from '../components/places/PlaceDetail';

function PlaceDetailContainer() {
    const [place, setPlace] = useState();
    const [loading, setLoading] = useState();

    const {id} = useParams();

    useEffect(() =>{
        getPlaceById(id)
        .then(setPlace)
        .finally(() => setLoading(false))
    }, [id]);

    if(loading) {
        return <h1>Loading...</h1>
    }
    return (
        <>
            <PlaceDetail place={place}/>
        </>
    )
}

export default PlaceDetailContainer;