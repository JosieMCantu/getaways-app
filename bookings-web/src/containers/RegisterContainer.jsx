import React, {useState} from 'react';
import Register from '../components/auth/Register';
import {registerUser} from '../services/authUtils';


const RegisterContainer = () => {

    const [userName, setUserName] = useState('');
    const [password, setPassword] = useState('');
    const [email, setEmail] = useState('');

    const handleChange = async ({target}) => {

        if(target.name === 'username') setUserName(target.value);
        if(target.name === 'password') setPassword(target.value);
        if(target.name === 'email') setEmail(target.value);
    }
    
    const handleSubmit = async (event) => {
        event.preventDefault();
        await registerUser(userName, password, email);

    }

    return (
        <>
           <Register username={userName} password={password} email={email} onChange={handleChange} onSubmit={handleSubmit}/> 
        </>
    )
}

export default RegisterContainer;
