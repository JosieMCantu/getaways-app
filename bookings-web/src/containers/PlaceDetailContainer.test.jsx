require('dotenv').config();
import React from 'react';
import {rest} from 'msw';
import {setupServer} from 'msw/node';
import {render, screen} from '@testing-library/react';
import {PlaceDetailContainer} from '../containers/PlaceDetailContainer';
import {MemoryRouter, Route} from 'react-router-dom';
import data from '../fixtures/data.json';

const server = setupServer(
    rest.get('http://localhost:7890/api/v1/places', (req, res, ctx) => {
        return res(ctx.json({
            "id": "6092e208e9e24bf5b66a86d0",
            "name": "Practical Suites",
            "slug": "practical-suites",
            "description": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
            "location": "10546 Jaquan Rest, West Shanon, South Carolina, 35257",
            "price_per_night": 530,
            "image": "http://placeimg.com/1080/800",
            "image_thumbnail": "http://placeimg.com/400/400",
            "max_guests": 2,
            "pet_friendly": true,
            "pool": true,
            "wifi": false,
            "createdAt": "2021-05-05T18:20:56.023Z",
            "updatedAt": "2021-05-05T18:20:56.023Z",
            "__v": 0
            }))
    })
)

describe('Getaways Container', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());

    it('renders the place detail container to the screen', async () => {

        render(<MemoryRouter initialEntries={['/6092e208e9e24bf5b66a86d0']}><Route path="/:id"><PlaceDetailContainer /></Route></MemoryRouter>)

        const ul = await screen.findByRole('list', {name: 'place'})
        expect(ul).toMatchSnapshot();

    })
})


