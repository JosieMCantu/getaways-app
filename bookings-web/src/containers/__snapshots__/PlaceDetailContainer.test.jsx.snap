// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getaways Container renders the place detail container to the screen 1`] = `
<ul
  aria-label="place"
>
  <li>
    <img
      alt="Practical Suites"
      src="http://placeimg.com/1080/800"
    />
    Name: 
    Practical Suites
    Description: 
    The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design
    Location: 
    10546 Jaquan Rest, West Shanon, South Carolina, 35257
    Price Per Night: 
    530
    Max Guest: 
    2
    Pet Friendly: 
    Pool: 
    wifi: 
  </li>
</ul>
`;
